### PR TITLE
 feat: ability to customize version string when building custom bins 

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: fedimint-server/src/config/mod.rs [hunk 1]
```diff
@@ -241,7 +241,7 @@ impl ServerConfig {
             modules: Default::default(),
         };
         let consensus = ServerConfigConsensus {
-            code_version: CODE_VERSION.to_string(),
+            code_version: version_hash,
```
Comment (dpc) [inline fedimint-server/src/config/mod.rs:244]: **note:** Importantly before two different custom versions of Fedimint would not fail the version check during dkg.

**Comment by elsirion** [comment]
I'm a bit confused what the use case is here.

**Review by elsirion — APPROVED** [review]

**Comment by elsirion** [comment]
Doctests failing

**Comment by dpc** [comment]
> So maybe I get it now: if I build a custom fedimintd, currently the version hash reported would be the one from the main fedimint dependency and not the custom repo? This allows overriding it?

Exactly.

**Pull request merged**